lib/core/attemptPrettier.ts

```diff
@@ -0,0 +1,52 @@
+import { alerts } from "./alerts";
+import { canResolvePrettier } from "./canResolvePrettier";
+
+import { format, resolveConfig } from "prettier";
+
+interface Prettier {
+  format: typeof format;
+  resolveConfig: typeof resolveConfig;
+}
+
+const isPrettier = (t: unknown): t is Prettier =>
+  t &&
+  typeof t === "object" &&
+  t !== null &&
+  "format" in t &&
+  typeof (t as Prettier).format === "function" &&
+  "resolveConfig" in t &&
+  typeof (t as Prettier).resolveConfig === "function";
+
+/**
+ * Try to load prettier and config from project to format input,
+ * fall back to input if prettier is not found or failed
+ *
+ * @param {string} input
+ */
+export const attemptPrettier = async (input: string) => {
+  if (!canResolvePrettier()) {
+    return input;
+  }
+
+  const prettier = require("prettier");
+  if (!isPrettier(prettier)) {
+    // doesn't look like prettier
+    return input;
+  }
+
+  try {
+    const config = await prettier.resolveConfig(process.cwd(), {
+      editorconfig: true
+    });
+    // try to return formatted output
+    return prettier.format(
+      input,
+      Object.assign({}, config, { parser: "typescript" })
```

Comment:
Since this is [compiled down to `es5`](https://github.com/skovy/typed-scss-modules/blob/6a1c05b1a50ac56d945b62c1d600afa8c6c033bf/tsconfig.json#L3) we can use es6 spread: 

```suggestion
      { ...config, parser: "typescript" }
```